nexus/db-queries/src/db/queries/vpc_subnet.rs

```diff
@@ -404,6 +410,81 @@ mod test {
             "Should be able to insert a VPC Subnet with the same ranges in a different VPC",
         );
 
+        // We shouldn't be able to insert a subnet if the IPv4 or IPv6 blocks overlap.
+        // Explicitly test for different CIDR masks with the same network address
+        let new_row = VpcSubnet::new(
+            other_other_subnet_id,
+            vpc_id,
+            make_id(&other_name, &description),
+            overlapping_ipv4_block_longer,
+            other_ipv6_block,
+        );
+        let err = db_datastore
+            .vpc_create_subnet_raw(new_row)
+            .await
+            .expect_err("Should not be able to insert VPC Subnet with overlapping IPv4 range {overlapping_ipv4_block_longer}");
```

Comment:
Could we wrap this line? It's annoying, but we'll need to do that manually, since `rustfmt` won't automatically split strings. Something like:

```rust
.expect_err(
    "Should not be able to insert VPC Subnet with \
    overlapping IPv4 range, {overlapping_ipv4_block_longer}"
);
```